### ISSUE TITLE
Evaluate bulk operations under large deployments

### ISSUE THREAD

**Issue by harshavardhana** [minio/minio#8711]
** not using the template on purpose **

```go
/// Bucket operations

// MakeBucket - make a bucket.
func (xl xlObjects) MakeBucketWithLocation(ctx context.Context, bucket, location string) error {
        // Verify if bucket is valid.
        if err := s3utils.CheckValidBucketNameStrict(bucket); err != nil {
                return BucketNameInvalid{Bucket: bucket}
        }
```

```
// MakeBucketLocation - creates a new bucket across all sets simultaneously
// even if one of the sets fail to create buckets, we proceed to undo a
// successful operation.
func (s *xlSets) MakeBucketWithLocation(ctx context.Context, bucket, location string) error {
        g := errgroup.WithNErrs(len(s.sets))

        // Create buckets in parallel across all sets.
        for index := range s.sets {
                index := index
                g.Go(func() error {
                        return s.sets[index].MakeBucketWithLocation(ctx, bucket, location)
                }, index)
        }
```

```
// MakeBucketWithLocation - creates a new bucket across all zones simultaneously
// even if one of the sets fail to create buckets, we proceed all the successful
// operations.
func (z *xlZones) MakeBucketWithLocation(ctx context.Context, bucket, location string) error {
        if z.SingleZone() {
                return z.zones[0].MakeBucketWithLocation(ctx, bucket, location)
        }

        g := errgroup.WithNErrs(len(z.zones))

        // Create buckets in parallel across all sets.
        for index := range z.zones {
                index := index
                g.Go(func() error {
                        return z.zones[index].MakeBucketWithLocation(ctx, bucket, location)
                }, index)
        }
```

Under large deployments such as 1000+ disks and each server with 100's of drives we are not optimally using the network operations, instead, we are performing an equal number of network operations - we can avoid these by doing bulk calls. ie we can optimize the situation by making only a single network call per server.

We should be able to do these optimizations for all chatty operations such as GetBucketInfo(), MakeBucket(), DeleteBucket() etc.

**Comment by harshavardhana** [comment]
@kannappanr I will be working on this as per @abperiasamy

**Comment by klauspost** [comment]
`DeleteMultipleObjects` is also really chatty and slow.

**Comment by harshavardhana** [comment]
Lowering priority as priority has moved Versioning support first.

**Comment by harshavardhana** [comment]
Refer to the bulk lock fix #8974

**Comment by harshavardhana** [comment]
Most of the bulk operations are fixed already with large changes in faster reloads, reusing all the formatted disks etc.

Closing this issue as sufficiently addressed for now.
